Fix Make build under Linux (Ubuntu)

ic opened the pull request:
- Add missing files to the list of source to build.
- Add the -fPIC flag under Linux (ignored when no necessary).

ic committed:
Fix Make build under Linux (Ubuntu)

* Add missing files to the list of source to build.

* Add the -fPIC flag under Linux (ignored when no necessary).

ic committed:
Fix Make build under Linux (Ubuntu)

* Add missing files to the list of source to build.

* Add the -fPIC flag under Linux (ignored when no necessary).

tensorflow-jenkins commented:
Can one of the admins verify this patch?

ic commented:
I signed it!

petewarden commented on tensorflow/contrib/makefile/Makefile:
```diff
@@ -451,6 +452,7 @@ $(wildcard tensorflow/core/common_runtime/gpu_device_factory.*)
 TF_CC_SRCS := $(filter-out $(CORE_CC_EXCLUDE_SRCS), $(CORE_CC_ALL_SRCS))
 # Add in any extra files that don't fit the patterns easily
 TF_CC_SRCS += tensorflow/core/common_runtime/gpu/gpu_tracer.cc
+TF_CC_SRCS += tensorflow/core/kernels/deep_conv2d.cc
```
Could you add this file to tensorflow/contrib/makefile/tf_op_files.txt instead? That is where the kernel files are typically added:
https://github.com/tensorflow/tensorflow/tree/master/tensorflow/contrib/makefile#fixing-makefile-issues

Thanks!

ic committed:
Refactor Makefile change to comply with guidance.

https://github.com/tensorflow/tensorflow/tree/master/tensorflow/contrib/makefile#fixing-makefile-issues

ic committed:
Refactor Makefile change to comply with guidance.

https://github.com/tensorflow/tensorflow/tree/master/tensorflow/contrib/makefile#fixing-makefile-issues

petewarden commented:
Thanks @ic! Those changes look good now, I appreciate the patch.

ic commented:
@petewarden Thank you for the link---I missed that part, sorry.

Change done and tested on Mac OS X (10.10) and Linux Ubuntu 16_04.

I did not find any note in the contribution guideline on rebasing or squashing commits, so I left these as-is. Please let me know if there's something necessary.

petewarden commented:
@ic I believe the commiter will squash the commits, so that should be no problem, thanks again!

ic commented:
@petewarden Ok, thanks.

One side note: When I run `tensorflow/contrib/makefile/gen_file_lists.sh`, I have four files modified, and pretty unrelated to the change in this PR.

    modified:   tensorflow/contrib/makefile/proto_text_cc_files.txt
    modified:   tensorflow/contrib/makefile/proto_text_pb_cc_files.txt
    modified:   tensorflow/contrib/makefile/proto_text_pb_h_files.txt
    modified:   tensorflow/contrib/makefile/tf_proto_files.txt

The contents remove and add `.cc` files and `.proto`.

I think this is unrelated, but perhaps an important cleaning stage. Another similar issue has just popped up with the dynamic RNN API, and I wonder if the `gen_file_lists` script could not help keeping in sync with Bazel more easily. Working on this, so more issues to come~

ic commented:
@tensorflow-jenkins test this please

martinwicke reviewed (approved):
Pete says it looks good.

martinwicke commented:
@ic can you rebase please?

ic commented:
@martinwicke Rebase worked locally, but I am surprised with the result here. Looking into it...

ic commented:
A diff between the fork's `HEAD` and the `HEAD` here looks good:

    diff --git a/tensorflow/contrib/makefile/Makefile b/tensorflow/contrib/makefile/Makefile
    index b3c39b8..7d0f8b3 100644
    --- a/tensorflow/contrib/makefile/Makefile
    +++ b/tensorflow/contrib/makefile/Makefile
    @@ -188,7 +188,6 @@ ifeq ($(HAS_GEN_HOST_PROTOC),true)
            LIBFLAGS += -L$(MAKEFILE_DIR)/gen/protobuf-host/lib
            export LD_LIBRARY_PATH=$(MAKEFILE_DIR)/gen/protobuf-host/lib
     endif
    -       CXXFLAGS += -fPIC
            LIBFLAGS += -Wl,--allow-multiple-definition -Wl,--whole-archive
            LDFLAGS := -Wl,--no-whole-archive
     endif

Sorry, I do not see what happened here.

There was mention of squashing the PR's commits. Do you want me to do it?

caisq commented:
@tensorflow-jenkins test this please.
